src/test/pages/Loans/borrow.test.tsx

```diff
@@ -0,0 +1,90 @@
+import '@testing-library/jest-dom';
+
+import App from '@/App';
+import { WRAPPED_TOKEN } from '@/config/relay-chains';
+import {
+  DEFAULT_BORROW_POSITIONS,
+  DEFAULT_IBTC,
+  DEFAULT_LEND_POSITIONS,
+  DEFAULT_POSITIONS,
+  mockBorrow,
+  mockGetBorrowPositionsOfAccount,
+  mockGetLendPositionsOfAccount
+} from '@/test/mocks/@interlay/interbtc-api/parachain/loans';
+
+import { render, userEvent, waitFor } from '../../test-utils';
+import { submitForm, withinModalTabPanel } from '../utils/loans';
+import { TABLES } from './constants';
+
+const path = '/lending';
+const tab = 'borrow';
+
+describe('Borrow Flow', () => {
+  beforeEach(() => {
+    mockGetBorrowPositionsOfAccount.mockReturnValue(DEFAULT_BORROW_POSITIONS);
+    mockGetLendPositionsOfAccount.mockReturnValue(DEFAULT_LEND_POSITIONS);
+  });
+
+  afterAll(() => {
+    mockGetBorrowPositionsOfAccount.mockReturnValue(DEFAULT_BORROW_POSITIONS);
+    mockGetLendPositionsOfAccount.mockReturnValue(DEFAULT_LEND_POSITIONS);
+  });
+
+  it('should be able to borrow', async () => {
+    await render(<App />, { path });
+
+    const tabPanel = withinModalTabPanel(TABLES.BORROW.POSITION, tab, 'IBTC');
+
+    userEvent.type(tabPanel.getByRole('textbox', { name: 'borrow amount' }), DEFAULT_IBTC.AMOUNT.SMALL);
+
+    await submitForm(tabPanel, 'borrow');
+
+    expect(mockBorrow).toHaveBeenCalledWith(WRAPPED_TOKEN, DEFAULT_IBTC.MONETARY.SMALL);
+  });
+
+  it('should not be able to borrow', async () => {
```

Comment:
I think these should be separated out, so that you have one test for each scenario (i.e. insufficient collateral, and no collateral)